src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs

```diff
@@ -77,6 +78,20 @@ public static ResourceType SetBicepResourceProperties(ResourceType resourceType,
             switch (bodyType)
             {
                 case ObjectType bodyObjectType:
+                    // Extra fields should be added here.
+                    // Deployments RP
+                    if (string.Equals(bodyObjectType.Name, TemplateWriter.NestedDeploymentResourceType, StringComparison.CurrentCultureIgnoreCase))
```

Comment:
Always favor Ordinal over CurrentCulture unless you explicitly want to be culture sensitive. In this case you could use:
`StringComparer.OrdinalIgnoreCase.Equals(bodyObjectType.Name, TemplateWriter.NestedDeploymentResourceType)`